src/io/json/read/deserialize.rs

```diff
@@ -43,13 +59,15 @@ fn build_extract(data_type: &DataType) -> Extract {
             Box::new(move |value| {
                 let integer = match value {
                     Value::Number(number) => Some(deserialize_int_single::<i64>(*number)),
-                    Value::Bool(number) => Some(i64::from(*number)),
+                    Value::Bool(number) => Some(Ok(i64::from(*number))),
                     _ => None,
                 };
-                integer.map(|integer| {
-                    let mut hasher = DefaultHasher::new();
-                    hasher.write(&integer.to_le_bytes());
-                    (hasher.finish(), value)
+                integer.and_then(|integer_result| {
+                    integer_result.ok().map(|integer| {
```

Comment:
This is not amazing, as bad float values will be converted to NULL. But it's how things currently are for other issues in this parsing code, and it's an improvement over panic-ing.